runner/src/server/services/notifyService.ts

```diff
@@ -31,12 +32,28 @@ export class NotifyService {
     this.logger = server.logger;
   }
 
-  parsePersonalisations(options: Personalisation): Personalisation {
+  escapeURLs(value: string): string {
+    const specialCharactersPattern = new RegExp(/\[([^\[\]]*)]\(([^\(\)]*)\)/g);
+    const matches = [...value.matchAll(specialCharactersPattern)];
```

Comment:
I think we can just do replaceAll for this -

```suggestion
    const specialCharactersPattern = new RegExp(/\[([^\[\]]*)]\(([^()]*)\)/, 'g');
    
        return value.replaceAll(specialCharactersPattern, '\\[$1\\]\\($2\\)');
```

worth adding some inline docs for this method as well though